cli/src/commands/describe.rs

```diff
@@ -65,39 +77,159 @@ pub(crate) fn cmd_describe(
+                let mut commit_builder = tx
+                    .mut_repo()
+                    .rewrite_commit(command.settings(), commit)
+                    .detach();
+                if commit_builder.description().is_empty() {
+                    commit_builder.set_description(command.settings().default_description());
+                }
+                if args.reset_author {
+                    let new_author = commit_builder.committer().clone();
+                    commit_builder.set_author(new_author);
+                }
+                let temp_commit = commit_builder.write_hidden()?;
+                Ok((commit.id(), temp_commit))
+            })
+            .try_collect()?;
+
+        if temp_commits.len() == 1 {
+            let (_, temp_commit) = &temp_commits[0];
+            let template = description_template(&tx, "", temp_commit)?;
+            let description = edit_description(&repo, &template, command.settings())?;
+
+            vec![(&commits[0], description)]
+        } else {
+            let ParsedBulkEditMessage {
+                descriptions,
+                missing,
+                duplicates,
+                unexpected,
+            } = edit_multiple_descriptions(&mut tx, &repo, &temp_commits, command.settings())?;
+            if !missing.is_empty() {
+                return Err(user_error(format!(
+                    "The description for the following commits were not found in the edited \
+                     message: {}",
+                    missing.join(", ")
+                )));
+            }
+            if !duplicates.is_empty() {
+                return Err(user_error(format!(
+                    "The following commits were found in the edited message multiple times: {}",
+                    duplicates.join(", ")
+                )));
+            }
+            if !unexpected.is_empty() {
+                return Err(user_error(format!(
+                    "The following commits were not being edited, but were found in the edited \
+                     message: {}",
+                    unexpected.join(", ")
+                )));
+            }
+
+            let commit_descriptions = commits
+                .iter()
+                .map(|commit| {
+                    let description = descriptions.get(commit.id()).unwrap().to_owned();
+                    (commit, description)
+                })
+                .collect();
+
+            commit_descriptions
         }
-        let temp_commit = commit_builder.write_hidden()?;
-        let template = description_template(&tx, "", &temp_commit)?;
-        edit_description(tx.base_repo(), &template, command.settings())?
     };
-    commit_builder.set_description(description);
 
-    if commit_builder.description() != commit.description() || args.reset_author {
-        commit_builder.write(tx.mut_repo())?;
+    // Filter out unchanged commits to avoid traversing descendants unnecessarily.
```

Comment:
nit: iirc, this isn't just for optimization, but not to rebase unchanged ancestors by `transform_descendants()`. And the use of `transform_descendants()` is to simplify predecessor chain.